darts/models/forecasting/tsmixer_model.py

```diff
@@ -0,0 +1,943 @@
+"""
+Time-Series Mixer (TSMixer)
+-------
+The inner layers (``nn.Modules``) and the ``TimeBatchNorm2d`` were provided by a PyTorch implementation
+of TSMixer: https://github.com/ditschuk/pytorch-tsmixer
+
+The License of pytorch-tsmixer v0.2.0 from https://github.com/ditschuk/pytorch-tsmixer/blob/main/LICENSE,
+accessed Thursday, March 21st, 2024:
+'The MIT License
+
+Copyright 2023 Konstantin Ditschuneit
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+associated documentation files (the “Software”), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial
+portions of the Software.
+'
+"""
+
+from __future__ import annotations
+
+import operator
+from collections.abc import Callable
+from functools import reduce
+from typing import Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from darts.logging import get_logger, raise_log
+from darts.models.components import layer_norm_variants
+from darts.models.forecasting.pl_forecasting_module import (
+    PLMixedCovariatesModule,
+    io_processor,
+)
+from darts.models.forecasting.torch_forecasting_model import MixedCovariatesTorchModel
+from darts.utils.torch import MonteCarloDropout
+
+MixedCovariatesTrainTensorType = Tuple[
+    torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor
+]
+
+logger = get_logger(__name__)
+
+
+def time_to_feature(x: torch.Tensor) -> torch.Tensor:
+    """Converts a time series torch.Tensor to a feature torch.Tensor."""
+    return x.permute(0, 2, 1)
+
+
+feature_to_time = time_to_feature
+
+
+class TimeBatchNorm2d(nn.BatchNorm1d):
+    def __init__(self, normalized_shape: tuple):
```

Comment:
```suggestion
    def __init__(self, normalized_shape: Tuple[int, int]):
```

and you have to `from typing import Tuple`

Reply:
You are right, I probably should also exchange the "|" operator with Unions.
Forgot about that.

Reply:
Updated